File: minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min = Integer.MAX_VALUE;
        int left = 0; int right = 0; int sum = 0;
        while (right < nums.length) {
            sum += nums[right];
            while (sum >= target) {
                min = Math.min(min, right - left + 1);
                sum -= nums[left];
                left++;
            }
            right++;
        }
        return min == Integer.MAX_VALUE ? 0 : min;
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        sum = 0
        shortest = float('inf')
        for right in range(len(nums)):
            sum += nums[right]
            if sum >= target:
                shortest = min(shortest, right - left + 1)
            while sum >= target:
                sum -= nums[left]
                left += 1
                if sum >= target:
                    shortest = min(shortest, right - left + 1)
        return shortest if shortest != float('inf') else 0
            